### PR TITLE
[sentry-native] Update to 0.4.17

### PR DESCRIPTION
**Pull request by AenBleidd**
Signed-off-by: Vitalii Koshura <lestat.de.lionkur@gmail.com>

### CODE REVIEW

File: ports/sentry-native/fix-0.4.16-build.patch [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ 		integrations/sentry_integration_qt.h
+ 	)
+ endif()
+-
+-if(SENTRY_BUILD_EXAMPLES)
+-	sentry_target_sources_cwd(sentry
+-	sentry_tracing.c
+-	sentry_tracing.h
+-	)
+-endif()
```
Comment (Thomas1664) [inline ports/sentry-native/fix-0.4.16-build.patch:24]: Why is this needed even when `SENTRY_BUILD_EXAMPLES` is set to `OFF` in `vcpkg_cmake_configure`?

File: ports/sentry-native/portfile.cmake [hunk 2]
```diff
@@ -1,7 +1,7 @@
 vcpkg_download_distfile(ARCHIVE
```
Comment (Thomas1664) [inline ports/sentry-native/portfile.cmake:1]: Why not use `vcpkh_from_github`?

File: ports/sentry-native/fix-0.4.16-build.patch (hunk 1 above)
Comment (AenBleidd) [reply ports/sentry-native/fix-0.4.16-build.patch:24]: Current version has an issue: https://github.com/getsentry/sentry-native/issues/701
If next release with this hotfix will be released soon - I'll update this PR and remove this patch.

File: ports/sentry-native/portfile.cmake (hunk 2 above)
Comment (AenBleidd) [reply ports/sentry-native/portfile.cmake:1]: This port uses some dependencies with version pinning. To simplify port build it's easier to do this way

File: ports/sentry-native/vcpkg.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "sentry-native",
-  "version-semver": "0.4.15",
+  "version-semver": "0.4.16",
```
Comment (LilyWangLL) [inline ports/sentry-native/vcpkg.json:3]:
```suggestion
  "version": "0.4.16",
```
This library isn't a semver library; it should be switched to `"version"`.

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

File: ports/sentry-native/fix-0.4.16-build.patch (hunk 1 above)
Comment (AenBleidd) [reply ports/sentry-native/fix-0.4.16-build.patch]: Not applicable anymore

File: ports/sentry-native/vcpkg.json (hunk 3 above)
Comment (AenBleidd) [reply ports/sentry-native/vcpkg.json]: Fixed

**Review by LilyWangLL — APPROVED** [review]

**Review by dg0yt — CHANGES_REQUESTED** [review]
- On x64-linux, there is a build warning about an unsed parameter `CRASHPAD_ZLIB_SYSTEM`. This should be resolved, at least via `MAYBE_UNUSED_VARIABLES`.
- There is a selection of a `SENTRY_BACKEND`, but none of the backends is a port dependency.
  Does it use vendored dependencies? OTOH there is patching for `find_dependency(crashpad)`.
- The CMake config is installed as `share/sentry-native/sentry-config.cmake`. I don't think this can work with the reported usage `find_package(sentry-native CONFIG)`: Directory name and package name must match.
  This should be fixed with `vcpkg_cmake_config_fixup(PACKAGE_NAME sentry ...)`.

**Comment by AenBleidd** [comment]
@dg0yt,

> * On x64-linux, there is a build warning about an unsed parameter `CRASHPAD_ZLIB_SYSTEM`. This should be resolved, at least via `MAYBE_UNUSED_VARIABLES`.

Fixed

> * There is a selection of a `SENTRY_BACKEND`, but none of the backends is a port dependency.
>   Does it use vendored dependencies? OTOH there is patching for `find_dependency(crashpad)`.

Yes, this is provided with sentry-native archive. It might be good to unlink this and use `find_dependency` once, but this is definitely out-of-scope of this PR.

> * The CMake config is installed as `share/sentry-native/sentry-config.cmake`. I don't think this can work with the reported usage `find_package(sentry-native CONFIG)`: Directory name and package name must match.
>   This should be fixed with `vcpkg_cmake_config_fixup(PACKAGE_NAME sentry ...)`.

There is already
```
vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/sentry)
```
Isn't this enough?

**Comment by dg0yt** [comment]
> > * The CMake config is installed as `share/sentry-native/sentry-config.cmake`. I don't think this can work with the reported usage `find_package(sentry-native CONFIG)`: Directory name and package name must match.
> >   This should be fixed with `vcpkg_cmake_config_fixup(PACKAGE_NAME sentry ...)`.
> 
> There is already
> 
> ```
> vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/sentry)
> ```
> 
> Isn't this enough?

No: `PACKAGE_NAME` defaults to `${PORT}` (i.e. `sentry-native`), but as you can see from `CONFIG_PATH` and config file name, the package name should be `sentry`.
(I think this should eventually be fixed in `vcpkg_cmake_config_fixup`, but this is out of scope for this port.)

**Comment by AenBleidd** [comment]
@dg0yt, ok, done

**Comment by BillyONeal** [comment]
@dg0yt Thanks for the review!

@AenBleidd Thanks for the update!
